packages/components/src/components/button/button.component.ts

```diff
@@ -0,0 +1,255 @@
+import { CSSResult, html, PropertyValueMap } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import styles from './button.styles';
+import { Component } from '../../models';
+import { BUTTON_COLORS, BUTTON_TYPE, BUTTON_VARIANTS, DEFAULTS } from './button.constants';
+import { getIconNameWithoutStyle, getIconSize, isValidIconSize, isValidPillSize } from './button.utils';
+
+/**
+ * button component, which ...
+ *
+ * @tagname mdc-button
+ *
+ * @slot default - This is a default/unnamed slot
+ *
+ * @cssprop --custom-property-name - Description of the CSS custom property
+ */
+class Button extends Component {
+  @property({ type: Boolean }) active: boolean = false;
+
+  @property({ type: Boolean }) disabled: boolean = false;
+
+  @property({ type: Boolean, attribute: 'soft-disabled' }) softDisabled: boolean = false;
+
+  @property({ type: String, attribute: 'aria-label' }) override ariaLabel: string = '';
+
+  @property({ type: String }) variant = DEFAULTS.VARIANT;
+
+  @property({ type: Number }) size = DEFAULTS.SIZE;
+
+  @property({ type: String }) color = DEFAULTS.COLOR;
+
+  @property({ type: Number }) override tabIndex = 0;
+
+  @state() private type = DEFAULTS.TYPE;
+
+  @state() private iconSize = 1;
+
+  constructor() {
+    super();
+    this.role = 'button';
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('click', this.handleClick);
+    this.addEventListener('keydown', this.handleKeyDown);
+    this.addEventListener('keyup', this.handleKeyUp);
+    this.addEventListener('focus', this.handleFocus);
+    this.addEventListener('blur', this.handleBlur);
+  }
+
+  public override disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('click', this.handleClick);
+    this.removeEventListener('keydown', this.handleKeyDown);
+    this.removeEventListener('keyup', this.handleKeyUp);
+    this.removeEventListener('focus', this.handleFocus);
+    this.removeEventListener('blur', this.handleBlur);
+  }
```

Comment:
This [article] suggests us to use javascript constructor for a single call events.
As the above events doesn't need to be re rendered / re attached again, we can follow this below approach:

```suggestion
  constructor() {
    super();
    this.role = 'button';
    this.handleButtonEvents();
  }

  private handleButtonEvents() {
    this.addEventListener('click', this.handleClick);
    this.addEventListener('keydown', this.handleKeyDown);
    this.addEventListener('keyup', this.handleKeyUp);
    this.addEventListener('focus', this.handleFocus);
    this.addEventListener('blur', this.handleBlur);
  }
```

Reg `removeEventListener`: we don't neeed to remove the events manually as JS will clean up when this component is no longer used.

Reply:
Do we need to create `handleButtonEvents`?
Can't we directly add them into the constructor code?

Reply:
It's always better to keep the constructor clean.
My suggestion is to have a separate method to register all event listeners at one place.